### PR TITLE
Improves Test Coverage and refactors deprecations out of the tests

### PR DESCRIPTION
**Pull request by bsbodden**
* bumps up the test coverage from 59% to 68%
* refactors calls in the tests to deprecated methods
* add @SupressWarnings where appropriate (non-Bean JSON test objects)
* organizes the import statements
* fixes test (delException) which was missing @Test annotation and was otherwise broken

### CODE REVIEW

**Review by sazzad16 — COMMENTED** [review]
@bsbodden Could you please *unformat* the unchanged codes, specially in `ClientTest.java`? This would really help the review process.

**Comment by bsbodden** [comment]
> @bsbodden Could you please _unformat_ the unchanged codes, especially in `ClientTest.java`? This would really help the review process.

Done, I rebase the offending commit. I think I opened it in Eclipse to use the clean imports feature and it reformatted that file.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 1]
```diff
@@ -80,7 +80,7 @@
         }
     }
 
-	private Pool<Jedis> client;
+	protected Pool<Jedis> client;
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:83]: Revealing an internal object increases the chance of it being abused. Keep this one private.

```suggestion
	private Pool<Jedis> client;
```

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public JReJSON(Pool<Jedis> jedis) {
      * @param str the reply string to "analyze"
      * @throws RuntimeException
      */
-    private static void assertReplyNotError(final String str) {
+    protected static void assertReplyNotError(final String str) {
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:116]: We can trust a few private methods.

```suggestion
    private static void assertReplyNotError(final String str) {
```

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 3]
```diff
@@ -122,7 +122,7 @@ private static void assertReplyNotError(final String str) {
      * Helper to check for an OK reply
      * @param str the reply string to "scrutinize"
      */
-    private static void assertReplyOK(final String str) {
+    protected static void assertReplyOK(final String str) {
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:125]: We can trust a few private methods.
```suggestion
    private static void assertReplyOK(final String str) {
```

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java [hunk 4]
```diff
@@ -51,12 +60,13 @@ public IRLObject() {
         }
     }
 
+    @SuppressWarnings("unused")
     private static class FooBarObject {
-        public String foo;
+		public String foo;
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/ClientTest.java:65]:
```suggestion
        public String foo;
```

File: src/test/java/com/redislabs/modules/rejson/StaticClientTest.java [hunk 5]
```diff
@@ -28,20 +28,27 @@
...
-import java.util.HashMap;
+import redis.clients.jedis.Jedis;
 
 public class StaticClientTest {
 
     /* A simple class that represents an object in real life */
+	@SuppressWarnings("unused")
     private static class IRLObject {
-        public String str;
+		public String str;
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/StaticClientTest.java:51]:
```suggestion
        public String str;
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 2 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java:116]: Making them private prevents me from testing them. I just make them package level instead.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 3 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java:125]: Making them private prevents me from testing them. I just make them package level instead.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 1 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java:83]: What do you think of adding an accessor for the underlying Jedis Pooled client? I'll add that to a patch commit.

Comment (sazzad16) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Skip testing this.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 2 above)
Comment (sazzad16) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Skip testing this.

**Comment by bsbodden** [comment]
@sazzad16 Thanks for review! I think I addressed all the requests. Let me know. Cheers!

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 3 above)
Comment (sazzad16) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Skip testing this.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 6]
```diff
@@ -108,12 +108,16 @@ public JReJSON(Pool<Jedis> jedis) {
         this.client = jedis;
     }
 
-    /**
+    public Pool<Jedis> getClient() {
+		return client;
+	}
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:113]: Skip testing this.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 7]
```diff
@@ -108,12 +108,16 @@ public JReJSON(Pool<Jedis> jedis) {
...
+
+	/**
      *  Helper to check for errors and throw them as an exception
      * @param str the reply string to "analyze"
      * @throws RuntimeException
      */
-    private static void assertReplyNotError(final String str) {
+    static void assertReplyNotError(final String str) {
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:120]: Skip testing this.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 8]
```diff
@@ -122,7 +126,7 @@ private static void assertReplyNotError(final String str) {
      * Helper to check for an OK reply
      * @param str the reply string to "scrutinize"
      */
-    private static void assertReplyOK(final String str) {
+    static void assertReplyOK(final String str) {
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:129]: Skip testing this.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 1 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Addressed

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 2 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Addressed, back to private

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 6 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Done

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 8 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: Done

**Review by sazzad16 — APPROVED** [review]

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 9]
```diff
@@ -81,34 +81,34 @@
...
-     * 
+     *
      * @param jedis bring your own Jedis pool
      */
     public JReJSON(Pool<Jedis> jedis) {
         this.client = jedis;
     }
 
-    /**
+	/**
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:111]:
```suggestion
    /**
```

Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:111]:
```suggestion
    /**
```

File: src/test/java/com/redislabs/modules/rejson/StaticClientTest.java [hunk 10]
```diff
@@ -50,6 +57,7 @@ public IRLObject() {
         }
     }
 
+	@SuppressWarnings("unused")
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/StaticClientTest.java:60]:
```suggestion
    @SuppressWarnings("unused")
```

File: src/test/java/com/redislabs/modules/rejson/StaticClientTest.java [hunk 11]
```diff
@@ -28,18 +28,25 @@
...
-import static junit.framework.TestCase.*;
+import com.google.gson.Gson;
 
-import java.util.HashMap;
+import redis.clients.jedis.Jedis;
 
 public class StaticClientTest {
 
     /* A simple class that represents an object in real life */
+	@SuppressWarnings("unused")
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/StaticClientTest.java:49]:
```suggestion
    @SuppressWarnings("unused")
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**